File: code/cpp/Code_2_04_2.cpp
#include <iostream>
using namespace std;

int main() {
	// 輸入
	int N, X, Y;
	cin >> N >> X >> Y;

	// 求解
	int cnt = 0;
	for (int i = 1; i <= N; i++) {
		if (i % X == 0 || i % Y == 0) cnt++; // mod 的計算請參考 2.2 節
	}

	// 輸出
	cout << cnt << endl;
	return 0;
}

File: code/python/Code_2_04_2.py
# 輸入
N, X, Y = map(int, input().split())

# 求出答案
cnt = 0
for i in range(1, N + 1):
	# mod 的計算請參考 2.2 節
	if i % X == 0 or i % Y == 0:
		cnt += 1

# 輸出
print(cnt)
